### PR TITLE
DOCS-468: Updated docs/README.md

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -7,16 +7,18 @@ Changelog category (leave one):
...
 
 
-Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
+Changelog entry (short description of added functionality):
 
 ...
 
 
-Detailed description (optional):
+Detailed description / Documentation:
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Detailed description / Documentation draft:
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -7,16 +7,18 @@ Changelog category (leave one):
...
 
-...
+Adding documentation, you much help users of a feature you have done. Documentation is highly required for all features that affect user experience.
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md]: let's keep `...` before this as well to make it more consistent

Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is highly required for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -7,16 +7,18 @@ Changelog category (leave one):
...
 ...
 
 
-Detailed description (optional):
+Detailed description / Documentation:
 
-...
+Adding documentation, you much help users of a feature you have done. Documentation is highly required for all features that affect user experience. 
+
+Read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide. It would help you to find the right form and words suitable for the audience of a new feature.
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,8 +1,16 @@
 # Contributing to ClickHouse
 
+ClickHouse is an open project, and you can contribute to it in many ways. You can help with ideas, code, or documentation. We appreciate any efforts that help us to make the project better.
+
+Thank you :-)
```
Comment (blinkov) [inline CONTRIBUTING.md]: smiles are inappropriate style for documentation itself and I think we'd better avoid it on such a visible place

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
+
 ## Technical info
```
Comment (blinkov) [inline CONTRIBUTING.md]: **I**nfo

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,8 +1,16 @@
...
-Developer guide for writing code for ClickHouse is published on official website alongside the usage and operations documentation:
-https://clickhouse.yandex/docs/en/development/architecture/
+
+You can find the "Contacts" section on the project's [website](https://clickhouse.yandex/). Select any way of communicating with us and discuss ideas.
```
Comment (blinkov) [inline CONTRIBUTING.md]: I'd remove this, it's not clear who is "us" and there are too many ways of communication

File: docs/README.md [hunk 7]
```diff
@@ -1,62 +1,218 @@
-# How to Contribute to ClickHouse Documentation
+# Contributing to ClickHouse Documentation
 
-ClickHouse uses the "documentation as code" approach, so you can edit Markdown files in this folder from the GitHub web interface. Alternatively, fork the ClickHouse repository, edit, commit, push, and open a pull request.
+## Why You Need to Document ClickHouse
 
-At the moment documentation is bilingual in English and Russian. Try to keep all languages in sync if you can, but this is not strictly required. There are people who are responsible for monitoring language versions and syncing them. If you add a new article, you should also add it to `toc_{en,ru,zh,ja,fa}.yaml` files with the pages index.
+The main reason is that ClickHouse is an open source project, and if you don't write the docs, nobody does. "Incomplete or Confusing Documentation" is the top complaint about open source software by the results of a [Github Open Source Survey](http://opensourcesurvey.org/2017/) of 2017. Documentation is highly valued but often overlooked. One of the most important contributions someone can make to an open source repository is a documentation update.
```
Comment (blinkov) [inline docs/README.md]: no newer surveys?

File: docs/README.md [hunk 8]
```diff
@@ -1,62 +1,218 @@
...
 
-The master branch is then asynchronously published to the ClickHouse official website:
+Many developers can say that the code is the best docs by itself, and they are right. But, ClickHouse is not a project for C++ developers. Most of its users don't know C++, and they can't read the code. ClickHouse is large enough to absorb almost any change without a noticeable trace. Nobody will find your very useful function, or an important setting, or a very informative new column in a system table if it is not referenced in the documentation.
```
Comment (blinkov) [inline docs/README.md]:
```suggestion
Many developers can say that the code is the best docs by itself, and they are right. But, ClickHouse is not a project for C++ developers. Most of its users don't know C++, and they can't understand the code quickly. ClickHouse is large enough to absorb almost any change without a noticeable trace. Nobody will find your very useful function, or an important setting, or a very informative new column in a system table if it is not referenced in the documentation.
```

File: docs/README.md [hunk 9]
```diff
@@ -1,62 +1,218 @@
...
-* In Russian: https://clickhouse.yandex/docs/ru/
-* In Chinese: https://clickhouse.yandex/docs/zh/
-* In Japanese: https://clickhouse.yandex/docs/ja/
-* In Farsi: https://clickhouse.yandex/docs/fa/
+If you want to help ClickHouse with documentation you can face, for example, the following questions:
 
-The infrastructure to build Markdown for publishing on the documentation website resides in the [tools](tools) folder. It has its own [README.md](tools/README.md) file with more details.
+- "I don't know how to write."
+    
+    We prepared some [recommendations](#what-to-write) for you.
```
Comment (blinkov) [inline docs/README.md]:
```suggestion
    We have prepared some [recommendations](#what-to-write) for you.
```

File: docs/README.md [hunk 10]
```diff
@@ -1,62 +1,218 @@
...
+- "I know what I want to write, but I don't know how to contribute to docs."
 
-## Target Audience
+    Here are some [tips](#how-to-contribute).
 
-When you write pretty much any text, the first thing you should think about is who will read it and which terms you should use for communicating with them.
+- "I don't know what to describe."
 
-ClickHouse can be directly used by all sorts of analysts and engineers. For generic parts of documentation (like the query language, tutorials or overviews), assume that the reader only has a basic technical background. For more technical sections (like articles that describe ClickHouse internals, guides for operating ClickHouse clusters, or rules for contributing to C++ code), you can use technical language and concepts.
+    Ask us, and we will provide you a choice.
```
Comment (blinkov) [inline docs/README.md]: how? and who's "us"?

File: docs/README.md [hunk 11]
```diff
@@ -1,62 +1,218 @@
...
+-  [How to Write Content for ClickHouse Documentation](#what-to-write)
+    - [Documentation for Different Audience](#target-audience)
+    - [Common Recommendations](#common-recommendations)
+    - [Description Templates](#templates)
+- [How to Build Documentation](#how-to-build-docs)
+
+
+<a name="clickhouse-docs"/>
+
+## What is the ClickHouse Documentation
```
Comment (blinkov) [inline docs/README.md]: I give up for now... it doesn't look lightweight

**Review by blinkov — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 11 above)
Comment (BayoNet) [reply docs/README.md]: There is no much text further :-).

File: docs/README.md (hunk 7 above)
Comment (BayoNet) [reply docs/README.md]: I couldn't find

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -7,16 +7,20 @@ Changelog category (leave one):
...
 
 ...
 
 
-Detailed description (optional):
+Detailed description / Documentation draft:
 
 ...
+
+By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is highly required for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.
```
Comment (blinkov) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.
```

**Pull request merged**